test/cookbooks/osl-git-test/recipes/databag.rb

```diff
@@ -0,0 +1,44 @@
+#
+# Cookbook:: osl-git-test
+# Recipe:: databag
+#
+# Copyright:: 2018, Oregon State University
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# This recipe tests git credentials from databags
+
+node.default['osl-git']['secrets_databag'] = 'osl-git'
+node.default['osl-git']['secrets_item']    = 'item1'
+
+# Testing default databag (from attributes)
+git_credentials 'root' do
+  notifies :sync, 'git[/root/test]', :immediately
+end
+
+# Testing git repo clone that requires credentials
+git '/root/test' do
+  repository 'https://git.osuosl.org/osuosl/test.git'
+end
+
+# Testing databag specified in properties
+git_credentials 'root' do
+  path '/root/.git-credentials-2'
+  secrets_databag 'osl-git'
+  secrets_item    'item2'
+end
```

Comment:
May I ask what the point is for this test? You're effectively doing resource cloning here since you're doing it for the root user here.

Reply:
It made sense earlier when it was an accumulator, but you're right that it doesn't anymore.

Reply:
Upon further thought, specifying a different databag item explicitly does cover cases where arguments are passed to the `git_credentials_secrets` library function (instead of defaulting to databag item in attributes).

I'm now specifying a different databag item as a different user in the tests, so there is no resource duplication.